Userland/Libraries/LibWeb/WebAssembly/WebAssembly.cpp

```diff
@@ -450,9 +451,20 @@ JS::Value to_js_value(JS::VM& vm, Wasm::Value& wasm_value)
         return JS::Value(wasm_value.to<double>().value());
     case Wasm::ValueType::F32:
         return JS::Value(static_cast<double>(wasm_value.to<float>().value()));
-    case Wasm::ValueType::FunctionReference:
-        // FIXME: What's the name of a function reference that isn't exported?
-        return create_native_function(vm, wasm_value.to<Wasm::Reference::Func>().value().address, "FIXME_IHaveNoIdeaWhatThisShouldBeCalled");
+    case Wasm::ValueType::FunctionReference: {
+        auto address = wasm_value.to<Wasm::Reference::Func>().value().address;
+        auto& cache = get_cache(realm);
+        auto* function = cache.abstract_machine().store().get(address);
+        ByteString name;
+        if (auto* wasm_function = function->get_pointer<Wasm::WasmFunction>()) {
+            auto index = *wasm_function->module().functions().find_first_index(address);
+            name = ByteString::formatted("func{}", index);
+        } else {
+            auto& host_function = function->get<Wasm::HostFunction>();
+            name = host_function.name();
+        }
```

Comment:
Please rewrite this using `visit` instead:
```suggestion
        auto name = function->visit(
            [&](Wasm::WasmFunction& wasm_function) {
                auto index = *wasm_function.module().functions().find_first_index(address);
                return ByteString::formatted("func{}", index);
            }, [](Wasm::HostFunction& host_function) {
                return host_function.name();
            });
```

Reply:
Done! Thanks you!